src/include/index/index_defs.h

```diff
@@ -57,6 +57,19 @@ enum class IndexKind { FlatL2, IVFFlat, Vamana, IVFPQ };
 
 enum class QueryType { FiniteRAM, InfiniteRAM };
 
+enum class IndexLoadStrategy {
+  // Used by Python to support `open_for_remote_query_execution`. In this case
+  // we don't want to load any data, we just want to load metadata so that we
+  // can call
+  // index.dimensions() and other metadata getters.
+  ONLY_METADATA,
+  // Load the index data as well as metadata.
+  DEFAULT,
```

Comment:
I would prefer to use more descriptive names i.e. `METADATA_ONLY, PQ_INDEX, PQ_INDEX_AND_RERANKING_VECTORS`